### PR TITLE
new filter for RB

### PR DESCRIPTION
**Pull request by eisbilir**

### CODE REVIEW

**Review by maxceem — CHANGES_REQUESTED** [review]
thanks, @eisbilir filtering works great.

There is just one typo in the error message. If I only filter by `workPeriods.isLastWeek=true` I get error `"Cannot set \"isFirstWeek\" without \"startDate\""`, though it suppose to be `isLastWeek` and `endDate` in the error message.

Also, there are a few suggestions on how to rephrase validation messages below.

File: src/services/ResourceBookingService.js [hunk 1]
```diff
@@ -902,11 +902,11 @@ searchResourceBookings.schema = Joi.object().keys({
         is: false,
         then: Joi.boolean().default(false),
         otherwise: Joi.boolean().valid(false).messages({
-          'any.only': 'Cannot set both "isFirstWeek" and "isLastWeek" to "true"'
+          'any.only': 'Cannot filter by both "isFirstWeek" and "isLastWeek" set to "true"'
         })
       }),
       otherwise: Joi.boolean().valid(false).messages({
-        'any.only': 'Cannot set "isFirstWeek" without "startDate"'
+        'any.only': 'Cannot filter by "isLastWeek" without "startDate"'
```
Comment (maxceem) [inline src/services/ResourceBookingService.js:909]: @eisbilir should it say `endDate` instead of `startDate`?

**Review by eisbilir — COMMENTED** [review]
2. workPeriods.isFirstWeek - should only return RB if the week which matches workPeriods.startDate is the first one in the RB
3. workPeriods.isLastWeek - should only return RB if the week which matches workPeriods.startDate is the last one in the RB

according to there, we match with startDate for both `workPeriods.isFirstWeek` and `workPeriods.isLastWeek`

**Review by maxceem — APPROVED** [review]
Thanks, @eisbilir all is perfect now.

**Pull request merged**